include/circt/Dialect/Moore/MIRStatements.td

```diff
@@ -0,0 +1,28 @@
+//===- MIRStatements.td - Moore MIR statements ops ---------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the ops for Moore MIR statements.
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// Assignment Statements
+//===----------------------------------------------------------------------===//
+
+def AssignOp : MIROp<"assign", []> {
```

Comment:
Should this have a verifier to check that the LHS and RHS types are equal? Or maybe we don't actually, since there is probably a stage in the pipeline where the LHS and RHS types don't agree, and a subsequent pass will insert cast operations. So maybe there will have to be multiple verifier *passes* that check invariants at different stages of the pipeline? (Like RHS castable to LHS at first, and after cast resolution RHS identical to LHS).

Reply:
Yeah I'm not sure yet either. But it's a good idea to start out conservative and check for equality.